### DISCUSSION TITLE
Input not working (docker)

### DISCUSSION THREAD

**Discussion by TheVexedGerman** [LizardByte/community#80] [Sunshine] [answered]
I'm trying to stream from a docker container. The docker host is Unraid 6.10, with the game host is running as a docker container with the nvidia runtime passed through. Kernel 5.15.46 with Nvidia driver v515.65.01 and a Quadro P2200 as the GPU. The docker is Debian 11 based.

I'm having issues with input not being handled on the host. According to the Sunshine debug log level the input is being received, it just doesn't appear to be actually relayed to the host. No error seen. `dmesg` shows Xbox controller, Logitech Wireless Mouse, and Keyboard passthrough being added when Sunshine is launched. Is there some extra library required?

Streaming using Steam works, but I'd prefer being able to use Sunshine with Moonlight due to client device compatibility.

**Comment by ReenigneArcher** [comment]
Are you using Games on Whales? We haven't added docker support quite yet, although it is in process.

**Reply by TheVexedGerman** [reply]
No, I've been using https://hub.docker.com/r/josh5/steam-headless and adding Sunshine to it. Just a normal install for now because I wanted to see if it works before I build a proper container. I'll try Games on Whales to see if I have more success with that project, it looks like they are focused on doing exactly on what I'm trying to accomplish.

I had kinda hoped it would just work in Docker since it is effectively running on the same system but with isolated processes. But I guess it isn't quite as easy as that.

**Reply by ReenigneArcher** [reply, marked as answer]
This is the PR that aims to add an official sunshine image. https://github.com/LizardByte/Sunshine/pull/325

I had to resolve some other things, which are done now, before continuing this work. Now I just need to find the time.

**Reply by thiago1255** [reply]
May i ask, whats the diference from the official images and just install sunshine in a container ?
What is the diference to be capable to use the xvfb instead a real X server ?

**Reply by ReenigneArcher** [reply]
Official, just means created by us.

https://github.com/LizardByte/Sunshine/blob/3239f2e1a460342bf6175fc886943fb419b41f2c/docker/ubuntu-24.04.dockerfile#L53-L106
